### PR TITLE
New option --store

### PR DESCRIPTION
**Pull request by jensklose**
New option '--store' implemented

RELEASE_NOTES=[Doc] Add an option '--store' to use as a prefix

Signed-off-by: Jens Klose <jens.klose@googlemail.com>

### CODE REVIEW

File: git-credential_test.go [hunk 1]
```diff
@@ -144,3 +146,103 @@ func TestGitCredentialHelper(t *testing.T) {
 	termio.Stdin = strings.NewReader("a")
 	assert.Error(t, act.Erase(c))
 }
+
+func TestGitCredentialHelperWithStoreFlag(t *testing.T) {
+	ctx := context.Background()
+	act := &gc{
+		gp: apimock.New(),
+	}
+	// require.NoError(t, act.gp.Set(ctx, "foo", &apimock.Secret{Buf: []byte("bar")}))
```
Comment (dominikschulz) [inline git-credential_test.go:155]: please remove this line if we don't need it.

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,4 @@
 git-credential-gopass
 dist/
 
+.vscode
```
Comment (dominikschulz) [inline .gitignore:20]: This looks odd in the diff. I think a final new line is missing.

File: git-credential.go [hunk 3]
```diff
@@ -135,6 +135,14 @@ func filter(ls []string, prefix string) []string {
 	return out
 }
 
+func composePath(c *cli.Context, cred *gitCredentials) string {
+	store := c.String("store") + "/"
+	if store == "/" {
+		store = ""
+	}
+	return store + "git/" + fsutil.CleanFilename(cred.Host) + "/" + fsutil.CleanFilename(cred.Username)
```
Comment (dominikschulz) [inline git-credential.go:143]: unrelated to this PR, but I think we need to rethink what cleanups `fsutil.CleanFilename` applies.

File: git-credential_test.go [hunk 4]
```diff
@@ -144,3 +146,103 @@ func TestGitCredentialHelper(t *testing.T) {
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getOptions(tt.args.c)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getOptions() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```
Comment (dominikschulz) [inline git-credential_test.go:243]: I prefer `assert.Equal` in the project. It uses `reflect.DeepEqual` under the hood, too.

**Review by dominikschulz — CHANGES_REQUESTED** [review]
Thanks a lot! This looks mostly very good.

File: git-credential.go (hunk 3 above)
Comment (jensklose) [reply git-credential.go:143]: /sign 
I'll test the windows way in gopass and then collect it in a new pull request agains the open issue.

**Review by dominikschulz — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**